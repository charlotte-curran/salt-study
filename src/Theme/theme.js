export const SaltTheme = {
  color: {
    dark: "#222324",
    blue: "#92b3d2",
    white: "rgba(255,255,255, 1)",
    orange: "#ff7a61",
    shadow: "#545353",
    address: "rgba(255, 255, 255, 0.4)",
    navbar: "rgba(34, 35, 36, 0.7)"
  },
  fonts: {
    futuraPt: '"futura-pt", sans-serif',
    raleway: '"raleway", sans-serif'
  }
};
