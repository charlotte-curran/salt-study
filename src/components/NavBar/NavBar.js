import React, { useState, useEffect } from "react";
import {
  Wrapper,
  SaltLink,
  Logo,
  LinkWrapper,
  LogoLink,
  SaltLabel,
  City
} from "./NavBar.styles";
import { useLocaleContext } from "../../Context/localeContext";
import SaltLogo from "../../images/saltlogo.svg";
import { useScrollContext } from "../../Context/scrollContext";

export const NavBar = props => {
  const [isCampusEpicenter, setIsCampusEpicenter] = useState(false);
  const [isCareerProgram, setIsCareerProgram] = useState(false);
  const [isHowItWorks, setIsHowItWorks] = useState(false);
  const [isOurTeam, setIsOurTeam] = useState(false);
  const [isDiversity, setIsDiversity] = useState(false);
  const { refObj, isVisibleModal } = props;

  const currPos = useScrollContext().y;
  const { locale, swapLocale } = useLocaleContext();

  useEffect(() => {
    const { secondWrapperRef, squareSixRef, thirdWrapperRef } = refObj;
    if (
      secondWrapperRef.current.offsetTop <= currPos &&
      currPos < squareSixRef.current.offsetTop
    ) {
      if (!isCampusEpicenter) {
        setIsCampusEpicenter(true);
      }
    } else {
      if (isCampusEpicenter) {
        setIsCampusEpicenter(false);
      }
    }
    if (
      squareSixRef.current.offsetTop <= currPos &&
      currPos <
        squareSixRef.current.offsetTop + squareSixRef.current.offsetHeight * 0.6
    ) {
      if (!isCareerProgram) {
        setIsCareerProgram(true);
      }
    } else {
      if (isCareerProgram) {
        setIsCareerProgram(false);
      }
    }
    if (
      squareSixRef.current.offsetTop + squareSixRef.current.offsetHeight * 0.6 <
        currPos &&
      currPos < thirdWrapperRef.current.offsetTop
    ) {
      if (!isHowItWorks) {
        setIsHowItWorks(true);
      }
    } else {
      if (isHowItWorks) {
        setIsHowItWorks(false);
      }
    }
    if (window.innerWidth >= 1000) {
      if (
        thirdWrapperRef.current.offsetTop <= currPos &&
        currPos <
          thirdWrapperRef.current.offsetTop +
            thirdWrapperRef.current.offsetHeight
      ) {
        if (!isOurTeam) {
          setIsOurTeam(true);
        }
      } else {
        if (isOurTeam) {
          setIsOurTeam(false);
        }
      }
    }
    if (window.innerWidth < 1000) {
      if (
        thirdWrapperRef.current.offsetTop <= currPos &&
        currPos <
          thirdWrapperRef.current.offsetTop +
            thirdWrapperRef.current.offsetHeight / 2
      ) {
        if (!isDiversity) {
          setIsDiversity(true);
        }
      } else {
        if (isDiversity) {
          setIsDiversity(false);
        }
      }
      if (
        thirdWrapperRef.current.offsetTop +
          thirdWrapperRef.current.offsetHeight / 2 <=
          currPos &&
        currPos <
          thirdWrapperRef.current.offsetTop +
            thirdWrapperRef.current.offsetHeight
      ) {
        if (!isOurTeam) {
          setIsOurTeam(true);
        }
      } else {
        if (isOurTeam) {
          setIsOurTeam(false);
        }
      }
    }
  }, [
    currPos,
    isCampusEpicenter,
    isCareerProgram,
    isDiversity,
    isHowItWorks,
    isOurTeam,
    refObj
  ]);

  return (
    <div>
      {!isVisibleModal ? (
        <Wrapper>
          <LogoLink
            onClick={() => {
              window.scrollTo(0, 0);
            }}
            href="#salt"
          >
            <Logo src={SaltLogo} />
          </LogoLink>
          <LinkWrapper>
            <SaltLink
              isCurrent={isCampusEpicenter}
              onClick={() => {
                setIsCampusEpicenter(true);
                window.scrollTo(
                  0,
                  refObj.secondWrapperRef.current.offsetTop + 1
                );
              }}
              href="#campus-epicenter"
            >
              campus epicenter
            </SaltLink>
            <SaltLink
              isCurrent={isCareerProgram}
              onClick={() => {
                setIsCareerProgram(true);
                window.scrollTo(0, refObj.squareSixRef.current.offsetTop + 1);
              }}
              href="#career-program"
            >
              career program
            </SaltLink>
            <SaltLink
              isCurrent={isHowItWorks}
              onClick={() => {
                setIsHowItWorks(true);
                window.scrollTo(
                  0,
                  refObj.squareSixRef.current.offsetTop +
                    refObj.squareSixRef.current.offsetHeight * 0.6 +
                    1
                );
              }}
              href="#how-it-works"
            >
              how does it work?
            </SaltLink>
            <SaltLink
              isCurrent={isOurTeam}
              onClick={() => {
                setIsOurTeam(true);
                window.scrollTo(
                  0,
                  refObj.thirdWrapperRef.current.offsetTop + 1
                );
              }}
              href="#diversity-in-tech"
            >
              diversity in tech
            </SaltLink>
            <SaltLink
              isCurrent={isOurTeam}
              onClick={() => {
                setIsOurTeam(true);
                window.scrollTo(
                  0,
                  refObj.thirdWrapperRef.current.offsetTop + 1
                );
              }}
              href="#our-team"
            >
              our team
            </SaltLink>
          </LinkWrapper>

          <SaltLabel>
            {isCampusEpicenter
              ? "campus epicenter"
              : isCareerProgram
              ? "career program"
              : isHowItWorks
              ? "how it works"
              : isOurTeam
              ? "our team"
              : isDiversity
              ? "diversity in tech"
              : null}
          </SaltLabel>
          <City
            onClick={() => {
              swapLocale(0);
            }}
          >
            {locale === 1
              ? "Stockholm"
              : locale === 2
              ? "Amsterdam"
              : locale === 3
              ? "Oslo"
              : ""}
          </City>
        </Wrapper>
      ) : null}
    </div>
  );
};

export default NavBar;
