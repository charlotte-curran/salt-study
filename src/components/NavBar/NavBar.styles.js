import styled from "styled-components";

export const Wrapper = styled.div`
  width: 100%;
  height: 35px;
  background: ${({ theme }) => theme.color.navbar};
  position: fixed;
  z-index: 3;
  color: ${({ theme }) => theme.color.white};
  padding: 0 20px;
  display: flex;
  align-items: center;
  box-sizing: border-box;
  justify-content: space-between;

  @media (min-width: 1000px) {
    height: 50px;
    padding: 0 40px;
    justify-content: space-between;
  }
`;

export const SaltLink = styled.a`
  font-size: 8px;
  font-family: ${({ theme }) => theme.fonts.raleway};
  color: ${({ theme }) => theme.color.white};
  text-transform: uppercase;
  text-decoration: ${({ isCurrent }) => (isCurrent ? "underline" : null)};
  letter-spacing: 3px;
  text-align: center;
  white-space: nowrap;
`;

export const City = styled.a`
  font-size: 10px;
  font-family: ${({ theme }) => theme.fonts.raleway};
  color: ${({ theme }) => theme.color.blue};
  text-transform: uppercase;
  letter-spacing: 2px;
  text-align: center;
  white-space: nowrap;
  font-weight: 600;
  cursor: pointer;

  @media (min-width: 1000px) {
    font-size: 12px;
  }
`;

export const SaltLabel = styled.span`
  font-size: 8px;
  font-family: ${({ theme }) => theme.fonts.raleway};
  color: ${({ theme }) => theme.color.white};
  text-transform: uppercase;
  letter-spacing: 3px;
  text-align: center;

  @media (min-width: 1000px) {
    display: none;
  }
`;

export const LinkWrapper = styled.div`
  display: none;

  @media (min-width: 1000px) {
    display: flex;
    align-items: center;
    justify-content: space-between;
    width: 60vw;
  }

  @media (min-width: 1300px) {
    display: flex;
    align-items: center;
    justify-content: space-between;
    width: 46vw;
  }
`;

export const LogoLink = styled.a`
  /* @media (min-width: 1000px) {
    left: -19%;
    position: relative;
  }

  @media (min-width: 1300px) {
    left: -25%;
    position: relative;
  } */
`;

export const Logo = styled.img`
  width: 63px;
  height: 13px;

  @media (min-width: 1000px) {
    width: 72px;
    height: 15px;
  }
`;
