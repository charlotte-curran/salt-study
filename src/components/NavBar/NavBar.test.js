import React from "react";
import { create } from "react-test-renderer";
import NavBar from "./NavBar";
import { ThemeProvider } from "styled-components";
import { SaltTheme } from "../../Theme/theme";
import ScrollProvider from "../../Context/scrollContext";
import LocaleProvider from "../../Context/localeContext";

describe("NavBar", () => {
  test("matches the snapshot", () => {
    const navBar = create(
      <LocaleProvider>
        <ScrollProvider>
          <ThemeProvider theme={SaltTheme}>
            <NavBar isVisibleModal={false} />
          </ThemeProvider>
        </ScrollProvider>
      </LocaleProvider>
    );
    expect(navBar.toJSON()).toMatchSnapshot();
  });
});
