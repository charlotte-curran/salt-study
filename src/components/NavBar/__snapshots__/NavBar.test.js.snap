// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar matches the snapshot 1`] = `
<div>
  <div
    className="sc-bdVaJa Sxtyk"
  >
    <a
      className="sc-EHOje ccJqxD"
      href="#salt"
      onClick={[Function]}
    >
      <img
        className="sc-bZQynM hrOpfu"
        src="saltlogo.svg"
      />
    </a>
    <div
      className="sc-ifAKCX cAQDKO"
    >
      <a
        className="sc-bwzfXH iBprWf"
        href="#campus-epicenter"
        onClick={[Function]}
      >
        campus epicenter
      </a>
      <a
        className="sc-bwzfXH iBprWf"
        href="#career-program"
        onClick={[Function]}
      >
        career program
      </a>
      <a
        className="sc-bwzfXH iBprWf"
        href="#how-it-works"
        onClick={[Function]}
      >
        how does it work?
      </a>
      <a
        className="sc-bwzfXH iBprWf"
        href="#diversity-in-tech"
        onClick={[Function]}
      >
        diversity in tech
      </a>
      <a
        className="sc-bwzfXH iBprWf"
        href="#our-team"
        onClick={[Function]}
      >
        our team
      </a>
    </div>
    <span
      className="sc-bxivhb gjSED"
    />
    <a
      className="sc-htpNat jgxKCt"
      onClick={[Function]}
    >
      
    </a>
  </div>
</div>
`;
